sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -260,10 +260,17 @@ public override string ClientRequestId
                 }
             }
 
-            // TryAddWithoutValidation overwrites non-list header values rather than appending them as lists in all cases.
-            // So a Remove call is not required when setting a header value.
-            protected internal override void SetHeader(string name, string value) =>
-                AddHeader(name, value);
+            protected internal override void SetHeader(string name, string value)
+            {
+                if (name.Equals(HttpHeader.Names.Authorization) && AuthenticationHeaderValue.TryParse(value, out var authHeader))
```

Comment:
Add a comment about why it's worth special-casing it. There are other headers that allow direct assignment.

Reply:
Is AuthenticationHeaderValue always allocated or are we adding an extra allocation?

Reply:
We could also eventually special case others that might be in the hot path.

Reply:
> Is AuthenticationHeaderValue always allocated or are we adding an extra allocation?

Yes, always allocated. There is no way around that though because `AuthenticationHeaderValue` is immutable.